### PR TITLE
feat(datasource/helm): add digest to release

### PR DESCRIPTION
**Pull request by scmeli**
## Changes

Add support for the digest of a chart archive within the helm datasource.

## Context

[discussion#23430](https://github.com/renovatebot/renovate/discussions/23430)

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
nope, doesn't work like this.

**Comment by scmeli** [comment]
> nope, doesn't work like this.

Thanks for the response. I updated the snapshot to fix the test cases. Can you elaborate if this is not what you meant? This change allows the regex manager to also update the digest for a helm datasource.

**Comment by scmeli** [comment]
@viceice Hi, can you provide more details? These changes work in my local environment so not sure what is broken. My use case involves updating the version/digest of a helm archive within a bazel http_file rule. This PR adds the digest to the helm datasource so I can use it with a regex manager. The digest is in the index.yaml which is already parsed by the helm datasource, but is not captured since it is missing from the HelmRelease type.

File: lib/modules/datasource/helm/index.ts [hunk 1]
```diff
@@ -71,6 +71,7 @@ export class HelmDatasource extends Datasource {
           releases: releases.map((release) => ({
             version: release.version,
             releaseTimestamp: release.created ?? null,
+            newDigest: release.digest ?? null,
```
Comment (viceice) [inline lib/modules/datasource/helm/index.ts:74]:
```suggestion
            newDigest: release.digest,
```
null isn't allowed on `newDigest` type

**Review by viceice — APPROVED** [review]

**Pull request merged**